modules/apps/layout/layout-reports-web/src/main/resources/META-INF/resources/js/LayoutReportsApp.js

```diff
@@ -12,33 +12,166 @@
+} from 'react';
 
 import BasicInformation from './components/BasicInformation';
 import EmptyLayoutReports from './components/EmptyLayoutReports';
 
-export default function ({context}) {
-	const {
-		assetsPath,
-		canonicalURLs,
-		configureGooglePageSpeedURL,
-		defaultLanguageId,
-		validConnection,
-	} = context;
-
-	return (
-		<>
-			<BasicInformation
-				canonicalURLs={canonicalURLs}
-				defaultLanguageId={defaultLanguageId}
-			/>
-
-			{validConnection || (
-				<EmptyLayoutReports
-					assetsPath={assetsPath}
-					configureGooglePageSpeedURL={configureGooglePageSpeedURL}
+const initialState = {
+	data: null,
+	error: null,
+	loading: false,
+};
+
+const dataReducer = (state, action) => {
+	switch (action.type) {
+		case 'LOAD_DATA':
+			return {
+				...state,
+				loading: true,
+			};
+
+		case 'SET_ERROR':
+			return {
+				...state,
+				error: action.error,
+				loading: false,
+			};
+
+		case 'SET_DATA':
+			return {
+				data: {
+					...action.data,
+				},
+				error: action.data?.error,
+				loading: false,
+			};
+
+		default:
+			return initialState;
+	}
+};
+
+export default function ({
+	isPanelStateOpen,
+	layoutReportsDataURL,
+	portletNamespace,
+}) {
+	const isMounted = useIsMounted();
+
+	const [state, dispatch] = useReducer(dataReducer, initialState);
+
+	const safeDispatch = (action) => {
+		if (isMounted()) {
+			dispatch(action);
+		}
+	};
+
+	const getData = (fetchURL) => {
+		safeDispatch({type: 'LOAD_DATA'});
+
+		fetch(fetchURL, {
+			method: 'GET',
+		})
+			.then((response) =>
+				response.json().then((data) =>
+					safeDispatch({
+						data,
+						type: 'SET_DATA',
+					})
+				)
+			)
+			.catch(() => {
+				safeDispatch({
+					error: Liferay.Language.get('an-unexpected-error-occurred'),
+					type: 'SET_ERROR',
+				});
+			});
+	};
+
+	const handlePageAuditData = useCallback(() => {
+		getData(layoutReportsDataURL);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [layoutReportsDataURL]);
+
+	useEffect(() => {
+		var layoutReportsPanelToggle = document.getElementById(
+			`${portletNamespace}layoutReportsPanelToggleId`
+		);
+
+		layoutReportsPanelToggle.addEventListener(
+			'mouseenter',
+			handlePageAuditData,
+			{once: true}
+		);
+		layoutReportsPanelToggle.addEventListener(
+			'focus',
+			handlePageAuditData,
+			{once: true}
+		);
+
+		var sidenavInstance = Liferay.SideNavigation.initialize(
+			layoutReportsPanelToggle
+		);
+
+		sidenavInstance.on('open.lexicon.sidenav', () => {
+			Liferay.Util.Session.set(
+				'com.liferay.layout.reports.web_layoutReportsPanelState',
+				'open'
+			);
+		});
+
+		sidenavInstance.on('closed.lexicon.sidenav', () => {
+			Liferay.Util.Session.set(
+				'com.liferay.layout.reports.web_layoutReportsPanelState',
+				'closed'
+			);
+		});
+
+		Liferay.once('screenLoad', () => {
+			Liferay.SideNavigation.destroy(layoutReportsPanelToggle);
+		});
+	}, [handlePageAuditData, portletNamespace]);
+
+	useLayoutEffect(() => {
```

Comment:
Can you explain why this is `useLayoutEffect` and not `useEffect`?

Reply:
I have no idea. So I think it's a mistake 😓